nucypher/blockchain/eth/sol/source/contracts/Issuer.sol

```diff
@@ -41,47 +48,68 @@ abstract contract Issuer is Upgradeable {
     uint16 public currentMintingPeriod;
 
     /**
-    * @notice Constructor sets address of token contract and coefficients for mining
-    * @dev Mining formula for one stake in one period
-    (totalSupply - currentSupply) * (lockedValue / totalLockedValue) * (k1 + allLockedPeriods) / k2
-    if allLockedPeriods > rewardedPeriods then allLockedPeriods = rewardedPeriods
+    * @notice Constructor sets address of token contract and coefficients for minting
+    * @dev Minting formula for one sub-stake in one period for the first phase
+    maxFirstPhaseReward * (lockedValue / totalLockedValue) * (k1 + allLockedPeriods) / k3
+    * @dev Minting formula for one sub-stake in one period for the seconds phase
+    (totalSupply - currentSupply) / k2 * (lockedValue / totalLockedValue) * (k1 + allLockedPeriods) / k3
+    if allLockedPeriods > maxRewardedPeriods then allLockedPeriods = maxRewardedPeriods
     * @param _token Token contract
     * @param _hoursPerPeriod Size of period in hours
-    * @param _miningCoefficient Mining coefficient (k2)
-    * @param _lockedPeriodsCoefficient Locked periods coefficient (k1)
-    * @param _rewardedPeriods Max periods that will be additionally rewarded
+    * @param _secondPhaseMintingCoefficient Minting coefficient for the second phase (k2)
+    * @param _lockingDurationCoefficient1 Numerator of he locking duration coefficient (k1)
+    * @param _lockingDurationCoefficient2 Denominator of the locking duration coefficient (k3)
+    * @param _maxRewardedPeriods Max periods that will be additionally rewarded
+    * @param _firstPhaseTotalSupply Total supply for the first phase
+    * @param _maxFirstPhaseReward Max possible reward for one period for all stakers in the first phase
```

Comment:
```suggestion
    * @param _firstPhaseMaxIssuance (Imax) Maximum number of new tokens minted per period during Phase 1. Imax = 1,002,509.479 NU tokens per period. See Equation 7 in Staking Protocol & Economics paper. 
```

Reply:
Don't want to include default value here, in contract. But name is perfect
✔️